### PR TITLE
D8CORE-330 Validate field name length

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Validate in the UI for field name length to avoid table hashing.

# Needed By (Date)
- asap

# Urgency
- Medium

# Steps to Test
1. you might have to enable ECK to test this.
1. create an eck entity type with a decently long machine name
1. create a field on that with a long name
1. validate the form throws an error.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
why you no codeclimate?

File: jumpstart_ui.module [hunk 1]
```diff
@@ -68,3 +68,31 @@ function jumpstart_ui_preprocess_patterns_overview_page(&$variables) {
...
+function jumpstart_ui_form_field_ui_field_storage_add_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['#validate'][] = 'jumpstart_ui_field_storage_add_validate';
+}
+
+/**
+ * Field storage add validation to prevent hashed table names.
+ *
+ * @see \Drupal\Core\Entity\Sql\DefaultTableMapping::generateFieldTableName()
+ */
+function jumpstart_ui_field_storage_add_validate(&$form, FormStateInterface $form_state) {
```
Comment (sherakama) [inline jumpstart_ui.module:86]: 🤔 I'm not sure this belongs here. The jumpstart_ui module is for visual changes and not site building practices but where to put it?

Comment (pookmish) [reply jumpstart_ui.module:86]: ya i figured since this will be everywhere. its either here or in `stanford_profile`

**Review by sherakama — CHANGES_REQUESTED** [review]
Please remove the alter from this PR and create a new module for it called `stanford_fields`

**Pull request closed without merging**